cli/Sources/TuistServer/Client/ServerAuthenticationController.swift

```diff
@@ -74,92 +88,175 @@ public struct ServerAuthenticationController: ServerAuthenticationControlling {
         #endif
     }
 
+    public func refreshToken(serverURL: URL, inSubprocess: Bool) async throws {
+        try await authenticationTokenRefreshingIfNeeded(
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: inSubprocess
+        )
+    }
+
     public func refreshToken(serverURL: URL) async throws {
         try await authenticationTokenRefreshingIfNeeded(
-            serverURL: serverURL, forceRefresh: true
+            serverURL: serverURL,
+            forceRefresh: true,
+            inSubprocess: true
         )
     }
 
     @discardableResult private func authenticationTokenRefreshingIfNeeded(
         serverURL: URL,
-        forceRefresh: Bool
+        forceRefresh: Bool,
+        inSubprocess: Bool
     ) async throws
         -> AuthenticationToken?
     {
         #if canImport(TuistSupport)
             Logger.current.debug("Refreshing authentication token for \(serverURL) if needed")
         #endif
-        return
-            try await CachedValueStore.current
-                .getValue(key: "token_\(serverURL.absoluteString)") {
-                    () -> (value: AuthenticationToken, expiresAt: Date?)? in
-                    guard let token = try await fetchTokenFromStore(serverURL: serverURL) else {
-                        return nil
-                    }
```

Comment:
This _must_ be preserved for the iOS and macOS app. Don't forget we're using this component for those apps, too. A ton of the refactoring here assumes you can touch the file system and that the CLI is available, we can assume neither of those.